packages/slider/index.tsx

```diff
@@ -0,0 +1,677 @@
+          return;
+        }
+        this.sliderElement.current.removeAttribute(name);
+      },
+      /**
+       * Returns the bounding client rect for the slider Element
+       */
+      computeBoundingRect: (): ClientRect | DOMRect => {
+        // new DOMRect is not IE11 compatible
+        this.rect = !this.sliderElement.current
+          ? {
+              bottom: 0,
+              height: 0,
+              left: 0,
+              right: 0,
+              top: 0,
+              width: 0,
+              x: 0,
+              y: 0,
+            }
+          : this.sliderElement.current.getBoundingClientRect();
+
+        return this.rect;
+      },
+      /**
+       * Returns the tab index of the slider Element
+       */
+      getTabIndex: (): number => this.props.tabIndex || 0,
+      // React handles events differently:
+      registerInteractionHandler: () => undefined,
+      deregisterInteractionHandler: () => undefined,
+      registerThumbContainerInteractionHandler: () => undefined,
+      deregisterThumbContainerInteractionHandler: () => undefined,
+      /**
+       * Registers an event handler on the body for a given event.
+       */
+      registerBodyInteractionHandler: (
+        evtType: K,
+        handler: SpecificEventListener<K>
+      ): void => {
+        document.body.addEventListener(evtType, handler);
+      },
+      /**
+       * Deregisters an event handler on the body for a given event.
+       */
+      deregisterBodyInteractionHandler: (
+        evtType: K,
+        handler: SpecificEventListener<K>
+      ): void => {
+        document.body.removeEventListener(evtType, handler);
+      },
+      /**
+       * Registers an event handler for the window resize event
+       */
+      registerResizeHandler: (
+        handler: SpecificEventListener<'resize'>
+      ): void => {
+        window.addEventListener('resize', handler);
+      },
+      /**
+       * Deregisters an event handler for the window resize event
+       */
+      deregisterResizeHandler: (
+        handler: SpecificEventListener<'resize'>
+      ): void => {
+        window.removeEventListener('resize', handler);
+      },
+      /**
+       * Emits a custom event MDCSlider:input from the root
+       */
+      notifyInput: (): void => {
+        const callback = this.props.onValueInput;
+        if (typeof callback === 'function') {
+          callback(this.foundation.getValue());
+        }
+      },
+      /**
+       * Emits a custom event MDCSlider:change from the root
+       */
+      notifyChange: (): void => {
+        const callback = this.props.onValueChange;
+        if (typeof callback === 'function') {
+          callback(this.foundation.getValue());
+        }
+      },
+      /**
+       * Sets a style property of the thumb container element to the passed value
+       */
+      setThumbContainerStyleProperty: (
+        propertyName: string,
+        value: string
+      ): void => {
+        this.setStylePropertyOn(
```

Comment:
do we know that this won't be overridden on a rerender? This updates the style property on the DOM Node. I'm not sure if it will persist.

Reply:
I just based it on the [tab-indicator](https://github.com/material-components/material-components-web-react/blob/master/packages/tab-indicator/index.tsx#L118).

I'll look at refactoring this.